server/rooms.ts

```diff
@@ -57,41 +57,41 @@ type Announcement = import('./chat-plugins/announcements').AnnouncementType;
 type Tournament = import('./tournaments/index').Tournament;
 
 export abstract class BasicRoom {
-	roomid: RoomID;
-	title: string;
-	parent: Room | null;
-	aliases: string[] | null;
-	users: {[userid: string]: User};
-	userCount: number;
-	type: 'chat' | 'battle' | 'global';
-	auth: {[userid: string]: string} | null;
+	readonly roomid: RoomID;
+	readonly parent: Room | null;
+	readonly aliases: string[] | null;
+	readonly users: {[userid: string]: User};
+	readonly type: 'chat' | 'battle' | 'global';
 	/**
 	 * Scrollback log. This is the log that's sent to users when
 	 * joining the room. Should roughly match what's on everyone's
 	 * screen.
 	 */
-	log: Roomlog | null;
-	game: RoomGame | null;
-	battle: RoomBattle | null;
-	active: boolean;
-	muteQueue: MuteEntry[];
+	readonly log: Roomlog | null;
+	readonly game: RoomGame | null;
+	readonly battle: RoomBattle | null;
+	readonly active: boolean;
```

Comment:
...should be actually modified by the server; I'm confused why this isn't a TypeScript error.